### ISSUE TITLE
Rethink naming

### ISSUE THREAD

**Issue by rklaehn** [n0-computer/quic-rpc#107]
Current naming is a bit confusing if you don't know QUIC. E.g. a connection is a thing that can give you bidi streams, but from TCP you would expect that a connection *is* a bidi stream.

So maybe `Connector` for the thing that has open and `Acceptor` for the thing that has accept?

Also, there are some old type names AcceptBiXXX or OpenBiXXX that need to be changed. The Bi can go, connections are always bidirectional.

**Comment by rklaehn** [comment]
The XBiY thing is taken care of in the map refactor PR.

**Comment by l-7-l** [comment]
I think this project is a great and promising project, covering the growth from modular monolith to microservices, and the name is actually good enough.
My suggestions:
1. Publish some introduction and performance comparison articles in communities such as reddit hacknews twitter This Week in Rust and other platforms for promotion
2. [ Why not unify the error handling of the server and client, and force the server to return Result?](https://github.com/google/tarpc/discussions/490#discussion-7451686)

**Comment by matheus23** [comment]
> the name is actually good enough

Yeah - this issue is more about the naming of *structs inside quic-rpc* and less about the name quic-rpc itself.

> Publish some introduction and performance comparison articles in communities such as reddit hacknews twitter This Week in Rust and other platforms for promotion

This is actually something *amazing* that you can contribute!
We might consider doing something like that too, but if you, personally, feel like this is needed and want to put some time into this, then that'd be a great contribution.

**Comment by l-7-l** [comment]
@matheus23 I'm sorry I'm not good at this. I'm new  backend developer and my English is very poor. 

_thanks for the great works and  have a niiiice day~~_
